### PR TITLE
Corrected filename and include static types

### PR DESCRIPTION
**Pull request by NumberPiOso**
continuing in line with #2128.

My first time contributing code to open source so criticism is encouraged.

- The name of the file is now compliant with python naming conventions
- Add static type as stated in contributing guidelines

### **Describe your change:**

* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.

### CODE REVIEW

**Comment by spamegg1** [comment]
Hi @NumberPiOso ,
It says you deleted the file, and rewrote it from scratch? It would have been better if you just edited it line by line so we could easily see the differences under "Files changed."

Also I see that you included type hints for some variables too. Like:
```python
user_input: str = input("Enter numbers separated by comma:\n").strip()
sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
```
etc.

I think [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) says to provide type hints only for the input parameters and return values of functions and nothing else, because it causes too much visual clutter.

Good luck and happy coding!

File: searches/simple_binary_search.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+"""
+Pure Python implementation of the binary search algorithm.
+
+For doctests run following command:
+python -m doctest -v simple_binary_search.py
```
Comment (cclauss) [inline searches/simple_binary_search.py:5]:
```suggestion
```

File: searches/simple_binary_search.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+or
```
Comment (cclauss) [inline searches/simple_binary_search.py:6]:
```suggestion
```

File: searches/simple_binary_search.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+python3 -m doctest -v simple_binary_search.py
+
+For manual testing run:
+python simple_binary_search.py
```
Comment (cclauss) [inline searches/simple_binary_search.py:10]:
```suggestion
python3 simple_binary_search.py
```

File: searches/simple_binary_search.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+"""
+from typing import List
+
+
+def binary_search(a_list: List[int], item: int) -> bool:
+    """
+    Pure python implementation of a binary search of a number is in a list.
```
Comment (cclauss) [inline searches/simple_binary_search.py:17]:
```suggestion
```
This was said already.

File: searches/simple_binary_search.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    >>> test_list = [0, 1, 2, 8, 13, 17, 19, 32, 42]
+    >>> print(binary_search(test_list, 3))
+    False
+    >>> print(binary_search(test_list, 13))
+    True
```
Comment (cclauss) [inline searches/simple_binary_search.py:23]: Please add tests for negative integers, floats, repeated numbers, letters.

File: searches/simple_binary_search.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    if a_list[midpoint] == item:
+        return True
+    if item < a_list[midpoint]:
+        return binary_search(a_list[:midpoint], item)
+    else:
+        return binary_search(a_list[midpoint + 1:], item)
+
+
+if __name__ == "__main__":
+    user_input: str = input("Enter numbers separated by comma:\n").strip()
```
Comment (cclauss) [inline searches/simple_binary_search.py:37]:
```suggestion
    user_input = input("Enter numbers separated by comma:\n").strip()
```
Python knows that input() returns a str.

File: searches/simple_binary_search.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
```
Comment (cclauss) [inline searches/simple_binary_search.py:38]:
```suggestion
    sequence = [int(item.strip()) for item in user_input.split(",")]
```

File: searches/simple_binary_search.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    target_str = input("Enter the number to be found in the list:\n").strip()
+    target: int = int(target_str)
```
Comment (cclauss) [inline searches/simple_binary_search.py:40]:
```suggestion
    target = int(target_str)
```

**Comment by NumberPiOso** [comment]
Hi @spamegg1 
> It says you deleted the file, and rewrote it from scratch? It would have been better if you just edited it line by line so we could easily see the differences under "Files changed."

It is because the other file was named **simple-binary-search.py** so I renamed it to **simple_binary_search.py**

I did it using *git mv*, is there any other way to do this in order for github to show us the differences?

> Also I see that you included type hints for some variables too. Like:
> 
> ```python
> user_input: str = input("Enter numbers separated by comma:\n").strip()
> sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
> ```
> 
> etc.
> 
> I think [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) says to provide type hints only for the input parameters and return values of functions and nothing else, because it causes too much visual clutter and Python is smart enough to figure it out. (There might be some exceptions to this but I don't know.)
> 

Totally agreed, I will correct it.

Thank you.

**Review by cclauss — APPROVED** [review]
LGTM.  It would be easier to review as two PRs.  One that changed the filename and the other that changed the file contents.

**Pull request merged**